pkg/extension/sumologicextension/extension.go

```diff
@@ -186,82 +180,33 @@ func (se *SumologicExtension) storeCollectorCredentials() error {
 	if err != nil {
 		return err
 	}
-	if err = os.WriteFile(path, encrypedCreds, 0600); err != nil {
-		return err
-	}
-	return nil
+	return fmt.Errorf("failed to save credentials file '%s': %w",
+		path,
+		os.WriteFile(path, encrypedCreds, 0600),
+	)
 }
 
-// ensureStoreCredentialsDir checks if directory to store credentials exists, if not try to create it.
+// ensureStoreCredentialsDir checks if directory to store credentials exists,
+// if not try to create it.
 func ensureStoreCredentialsDir(path string) error {
 	if _, err := os.Stat(path); err != nil {
-		err := os.Mkdir(path, 0600)
-		if err != nil {
+		if err := os.Mkdir(path, 0600); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// createHash creates key used for credentials encryption.
-func createHash(key string) (string, error) {
-	hasher := md5.New()
-	if _, err := hasher.Write([]byte(key)); err != nil {
-		return "", err
-	}
-	return hex.EncodeToString(hasher.Sum(nil)), nil
-}
-
-// encrypt encrypts collector credentials with AES using the passphrase
-func encrypt(data []byte, passphrase string) ([]byte, error) {
-	hash, err := createHash(passphrase)
-	if err != nil {
-		return nil, err
-	}
-	block, err := aes.NewCipher([]byte(hash))
-	if err != nil {
-		return nil, err
-	}
-	gcm, err := cipher.NewGCM(block)
-	if err != nil {
-		return nil, err
-	}
-	nonce := make([]byte, gcm.NonceSize())
-	if _, err = io.ReadFull(rand.Reader, nonce); err != nil {
-		return nil, err
-	}
-	ciphertext := gcm.Seal(nonce, nonce, data, nil)
-	return ciphertext, nil
-}
-
-// decrypt decrypts collector credentials encrypted with AES passphrase.
-func decrypt(data []byte, passphrase string) ([]byte, error) {
-	hash, err := createHash(passphrase)
-	if err != nil {
-		return nil, err
-	}
-	key := []byte(hash)
-	block, err := aes.NewCipher(key)
-	if err != nil {
-		return nil, err
-	}
-	gcm, err := cipher.NewGCM(block)
-	if err != nil {
-		return nil, err
-	}
-	nonceSize := gcm.NonceSize()
-	if nonceSize > len(data) {
-		return nil, fmt.Errorf("unable to decrypt credentials")
-	}
-	nonce, ciphertext := data[:nonceSize], data[nonceSize:]
-	plaintext, err := gcm.Open(nil, nonce, ciphertext, nil)
-	if err != nil {
-		return nil, err
+func (se *SumologicExtension) register(ctx context.Context) error {
+	type OpenRegisterRequestPayload struct {
```

Comment:
Why this is moved here? I would prefer to keep structs outside functions

Reply:
Moved it inside the registration in order to limit the scope because it's only used inside the `register` func.

I can leave it where it was if that's more readable 🤷